Frameworks/ImageIO/CGImageSource.mm

```diff
@@ -489,21 +1370,109 @@ size_t CGImageSourceGetCount(CGImageSourceRef isrc) {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes A limited subset of source container properties are supported currently  
 */
 CFDictionaryRef CGImageSourceCopyProperties(CGImageSourceRef isrc, CFDictionaryRef options) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!isrc);
+    NSData* imageData = ((ImageSource*)isrc).data;
+    RETURN_NULL_IF(!imageData);
+
+    NSMutableDictionary *properties = [[NSMutableDictionary alloc] init]; 
+    [properties setObject:[NSNumber numberWithInt:[imageData length]] forKey:(id)kCGImagePropertyFileSize];
+
+    // Look for container-level metadata if the file is a GIF.
```

Comment:
thanks; can you please add a bit more info about that to this comment?

---
In reply to: [60331063](https://github.com/Microsoft/WinObjC/pull/445#discussion_r60331063) [](ancestors = 60331063)